### DISCUSSION TITLE
How to reduce js bundle size ?

### DISCUSSION THREAD

**Discussion by SujayPrabhu96** [ant-design/ant-design#39017] [Q&A]
Hi folks, how to reduce js bundle size when using antd ? We are on a gatsby application that uses antd design and when looking into performance metrics from lighthouse found that antd includes all the packages in bundle thus increasing unnecessary js code. I have used [gatsby-plugin-import](https://www.gatsbyjs.com/plugins/gatsby-plugin-import/) to import only the component but I don't see any changes in bundle size

Versions
- antd: 4.12.2
- gatsby: 4.23.0

**Comment by vagusX** [comment]
try `[babel-plugin-import](https://www.npmjs.com/package/babel-plugin-import)` for antd v4?

**Reply by SujayPrabhu96** [reply]
@vagusX `gatsby-plugin-import` uses `babel-plugin-import`. But still I can see all the rc-* packages, moment js in bundle. Is there any way we can exclude these from the bundle ? 
For eg: `rc-field-form`, `moment` is included in the bundle which is a dependency of `antd`
